feat(publish): impl binary cache check retry (#239)
flox/flox: crates/flox/doc/flox-publish.md

    a different transport. For example, to upload packages
    to a (writable, authenticated) `s3://` URL,
    but download these packages from an (read-only,
    unauthenticated) `https://cache.floxdev.com endpoint`.

    If not provided the `--public-cache-url` argument will default to
    the `public_cache_url` config value,
    or same value as provided for the `--cache-url` argument.

`[ --retry <number> ]`

> :   Number of retries if the binary was not found in the cache at the first time

Reviewer: ```suggestion
:   Number of retries if the binary is not found in the cache
```
